optimizely/odp/odp_manager.py

```diff
@@ -29,55 +29,59 @@
 class OdpManager:
     """Orchestrates segment manager, event manager and odp config."""
 
-    def __init__(self,
+    def __init__(
+        self,
         disable: bool,
-        cache_size: int,
-        cache_timeout_in_sec: int,
+        segments_cache: OptimizelySegmentsCache,
         segment_manager: Optional[OdpSegmentManager] = None,
         event_manager: Optional[OdpEventManager] = None,
-        logger: Optional[optimizely_logger.Logger] = None) -> None:
+        logger: Optional[optimizely_logger.Logger] = None
+    ) -> None:
 
         self.enabled = not disable
-        self.cache_size = cache_size
-        self.cache_timeout_in_sec = cache_timeout_in_sec
         self.odp_config = OdpConfig()
         self.logger = logger or optimizely_logger.NoOpLogger()
 
         self.segment_manager = None
         self.event_manager = None
 
         if not self.enabled:
+            self.logger.info('ODP is disabled.')
             return
 
         if segment_manager:
             segment_manager.odp_config = self.odp_config
             self.segment_manager = segment_manager
         else:
             self.segment_manager = OdpSegmentManager(self.odp_config,
-                                                     LRUCache(cache_size, cache_timeout_in_sec),
+                                                     segments_cache,
                                                      ZaiusGraphQLApiManager(logger), logger)
         if event_manager:
             event_manager.odp_config = self.odp_config
             self.event_manager = event_manager
         else:
             self.event_manager = OdpEventManager(self.odp_config, logger)
 
-    def fetch_qualified_segments(self, user_id: str, options: list[str]) -> None:
-        if not self.enabled:
+        self.event_manager.start()
+
+    def fetch_qualified_segments(self, user_id: str, options: list[str]) -> Optional[list[str]]:
+        if not self.enabled or not self.segment_manager:
             raise optimizely_exception.OdpNotEnabled(Errors.ODP_NOT_ENABLED)
```

Comment:
I see queryql-api-manager returns null for all errors while you raises exception at this level. Consistency ?

Reply:
done